Microsoft.Toolkit.Uwp.UI.Animations/ConnectedAnimations/Connected.cs

```diff
@@ -0,0 +1,426 @@
+// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media.Animation;
+
+namespace Microsoft.Toolkit.Uwp.UI.Animations
+{
+    /// <summary>
+    /// A helper class that allows Connected Animations to be enabled through XAML
+    /// </summary>
+    /// <seealso cref="ConnectedAnimationService"/>
+    public static class Connected
+    {
+        private static List<ConnectedAnimationProperties> _connectedAnimationsProps = new List<ConnectedAnimationProperties>();
+        private static Dictionary<string, ConnectedAnimationProperties> _previousPageConnectedAnimationProps = new Dictionary<string, ConnectedAnimationProperties>();
+        private static Dictionary<UIElement, List<UIElement>> _coordinatedAnimationElements = new Dictionary<UIElement, List<UIElement>>();
+
+        private static Frame _navigationFrame;
+
+        /// <summary>
+        /// Gets or sets the <see cref="Frame"/> hosting the pages where the Connected Animation will run
+        /// Set this value if the frame hosting your pages is different than the root frame of the Window
+        /// </summary>
+        public static Frame NavigationFrame
+        {
+            get
+            {
+                return _navigationFrame;
+            }
+
+            set
+            {
+                if (_navigationFrame != null)
+                {
+                    _navigationFrame.Navigating -= NavigationFrame_Navigating;
+                    _navigationFrame.Navigated -= NavigationFrame_Navigated;
+                }
+
+                _navigationFrame = value;
+
+                if (!AnimationBase.IsCreatorsUpdateOrAbove)
+                {
+                    return;
+                }
+
+                _connectedAnimationsProps.Clear();
+                _previousPageConnectedAnimationProps.Clear();
+                _coordinatedAnimationElements.Clear();
+
+                _navigationFrame.Navigating += NavigationFrame_Navigating;
+                _navigationFrame.Navigated += NavigationFrame_Navigated;
```

Comment:
Event handlers in a static class can cause huge memory leaks. This needs to be rethinked, or the very least use weak event handlers.
In general it seems really odd that this is a static, and there can be only one navigation frame.

Reply:
Not sure that is the case here. In every one of these cases, the static class is the subscriber, so the objects can still be garbage collected regardless if there is an event handler registered in the static class.

Reply:
Sorry. It was late. I was tired.